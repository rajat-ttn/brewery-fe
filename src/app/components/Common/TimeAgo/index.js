import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';

class TimeAgo extends Component {
    constructor(props){
        super(props);
        this.state = {
            secondsElapsed: 0,
        };
    }
    tick = () => {
        this.setState({secondsElapsed: this.state.secondsElapsed + 1});
    }

    componentDidMount () {
        this.interval = setInterval(this.tick, 30000);
    }
    componentWillUnmount () {
        clearInterval(this.interval);
    }

    render() {
        const { beerContentDetail, beerList } = this.props;
        let lastTimeUpdated = '';
        beerList.filter(({ containerId,lastUpdatedTime }) => {
            if (beerContentDetail.containerId === containerId) {
                lastTimeUpdated = lastUpdatedTime;
            }
        });
        const lastTime = moment(lastTimeUpdated).fromNow();
        return (
            <small className="italic">
                Last updated: {lastTime}
            </small>
        );
    }
}

TimeAgo.propTypes = {
    beerList: PropTypes.array.isRequired,
    beerContentDetail: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
    return  ({
        beerList: state.beerList,
    })
};

export default connect(mapStateToProps, null)(TimeAgo);
