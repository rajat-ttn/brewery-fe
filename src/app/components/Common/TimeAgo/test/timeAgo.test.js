import React from 'react';
import { shallow, mount } from 'enzyme';
import TimeAgo from '../index';
import {TimeAgoContainer} from '../index';
import {mapStateToProps } from '../index';
import { connect } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('show last temperature correctly', () => {
    const initialState = {};
    const mockStore = configureStore();
    let TimeAgoComponent;
    let store = {
        beerList: [{beerType : "Aler Beer1", containerId: 1, id: 1, currentTemperature: 6, lastUpdatedTime: '', tempRange: [4,6] }, {beerType : "Aler Beer2", containerId: 2, id: 2, currentTemperature: 3, lastUpdatedTime: '', tempRange: [4,6] }],
        beerContentDetail: {beerType : "Aler Beer", containerId: 1, id: 1, currentTemperature: 6, lastUpdatedTime: '', tempRange: [4,6] }
    };
    const beerContentDetail= {beerType : "Aler Beer", containerId: 1, id: 1, currentTemperature: 6, lastUpdatedTime: '', tempRange: [4,6] };
    const beerList= [{beerType : "Aler Beer1", containerId: 1, id: 1, currentTemperature: 6, lastUpdatedTime: '', tempRange: [4,6] }, {beerType : "Aler Beer2", containerId: 2, id: 2, currentTemperature: 3, lastUpdatedTime: '', tempRange: [4,6] }];
    const dispatch= jest.fn();

    beforeEach(() => {
        store = mockStore(initialState);
        TimeAgoComponent = shallow(<TimeAgo store={store} beerContentDetail={beerContentDetail}  />);
    });


    it('last temperature update', () => {
        expect(TimeAgoComponent).toMatchSnapshot();
    });

    it('should set state', () => {
        const TimeAgoContainerComponent = shallow(<TimeAgoContainer beerList={beerList}  beerContentDetail={beerContentDetail}  />);
        expect(TimeAgoContainerComponent.state()).toEqual({ secondsElapsed: 0 });
    });

    it('should trigger tick event', () => {
        it('clicks it', () => {
            const timeAgoContainerInstance = TimeAgoContainer.instance();
            const tick = jest.spyOn(timeAgoContainerInstance.tick, 'tick');
            expect(tick).toHaveBeenCalled();
            expect(timeAgoContainerInstance.state()).toEqual({secondsElapsed: this.state.secondsElapsed + 1});

        })
    });

});