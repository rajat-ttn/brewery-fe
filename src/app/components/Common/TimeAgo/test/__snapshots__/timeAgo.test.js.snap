// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show last temperature correctly last temperature update 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TimeAgo)
    beerContentDetail={
        Object {
            "beerType": "Aler Beer",
            "containerId": 1,
            "currentTemperature": 6,
            "id": 1,
            "lastUpdatedTime": "",
            "tempRange": Array [
              4,
              6,
            ],
          }
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "beerContentDetail": Object {
        "beerType": "Aler Beer",
        "containerId": 1,
        "currentTemperature": 6,
        "id": 1,
        "lastUpdatedTime": "",
        "tempRange": Array [
          4,
          6,
        ],
      },
      "beerList": Array [],
      "dispatch": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "beerContentDetail": Object {
          "beerType": "Aler Beer",
          "containerId": 1,
          "currentTemperature": 6,
          "id": 1,
          "lastUpdatedTime": "",
          "tempRange": Array [
            4,
            6,
          ],
        },
        "beerList": Array [],
        "dispatch": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
